### ISSUE TITLE
Easier support for insecure requests?

### ISSUE THREAD

**Issue by vincentjames501** [RutledgePaulV/clj-okhttp#10]
Frameworks like clj-http make it pretty easy to hit services with self signed certs for example. Would you consider accepting a PR that adds an `:insecure?` flag to the client and automatically configures the necessary trust managers/ssl socket factories?

**Comment by RutledgePaulV** [comment]
While I know this can be helpful at times I'm not particularly interested in helping other developers bypass security features easily. Personally I think they should go through the trouble to add CAs to their trust store if they're going to do self-signed certs.

**Comment by vincentjames501** [comment]
@RutledgePaulV , I understand your point, though I do disagree. It's a giant pain to add CAs to trust stores especially if all you want to do is some REPL hacking or some scripting. In fact, I wanted to test some xml stuff out with this client and when querying our internal nexus/npm repositories I ran into it and I would literally have to burn 1hr of my time figuring out how to build silly trust managers as that's something I tend to only do once per year and hate my life doing it 🙂. I seem to use the `-k` option in cURL at least monthly. IMO, this is super easy to add as I did in #10 and it's clear when using it that it is `insecure` and therefore not recommended to use in production (and can be documented as such). If you're strongly against it, I'd recommend documenting a recipe for people to copy-paste.
